test/controllers/DisassociateDucrControllerSpec.scala

```diff
@@ -16,72 +16,140 @@
+      route(app, FakeRequest(GET, uri).withHeaders(headers.toSeq: _*).withCSRFToken).get
   }
 
-  "Disassociate Ducr Controller" should {
+  "Disassociate Ducr Controller on POST" when {
 
-    "return 200 for get request" in new SetUp {
+    "provided with correct data" should {
 
-      val result = route(app, getRequest(uri)).get
+      "return SeeOther code" in new Test {
 
-      status(result) must be(OK)
-    }
+        status(routePost()) must be(SEE_OTHER)
+      }
 
-    "return BadRequest for incorrect form" in new SetUp {
+      "call SubmissionService" in new Test {
 
-      val incorrectForm: JsValue = JsObject(
-        Map(
-          "ducr" -> JsString("abc")
+        routePost().futureValue
+
+        verify(submissionServiceMock).submitDucrDisassociation(meq(DisassociateDucr(correctUcr)))(
+          any(),
+          any()
         )
-      )
+      }
 
-      val result = route(app, postRequest(uri, incorrectForm)).get
+      "redirect to confirmation page" in new Test {
 
-      status(result) must be(BAD_REQUEST)
-      verifyZeroInteractions(mockSubmissionService)
+        redirectLocation(routePost()) must be(Some(routes.DisassociateDucrConfirmationController.displayPage().url))
+        verify(submissionServiceMock).submitDucrDisassociation(meq(DisassociateDucr(correctUcr)))(
+          any(),
+          any()
+        )
+      }
     }
 
-    "redirect to confirmation for correct form" in new SetUp {
-      given(mockSubmissionService.submitDucrDisassociation(anyString(), anyString())(any[HeaderCarrier], any[ExecutionContext])).willReturn(successful(():  Unit))
+    "provided with incorrect data" should {
 
-      val correctForm: JsValue =
-        JsObject(
-          Map(
-            "ducr" -> JsString("8GB12345612345612345")
-          )
-        )
+      "return BadRequest code" in new Test {
 
-      val result = route(app, postRequest(uri, correctForm)).get
+        status(routePost(body = incorrectForm)) must be(BAD_REQUEST)
+      }
 
-      status(result) must be(SEE_OTHER)
-      redirectLocation(result) must be(Some(routes.DisassociateDucrConfirmationController.displayPage().url))
-      verify(mockSubmissionService).submitDucrDisassociation(anyString, refEq("8GB12345612345612345"))(any[HeaderCarrier], any[ExecutionContext])
+      "not call SubmissionService" in new Test {
+
+        routePost(body = incorrectForm).futureValue
+
+        verifyZeroInteractions(submissionServiceMock)
+      }
     }
+
+    "SubmissionService returns status other than Accepted" should {
+      "return InternalServerError code" in new Test {
+        when(submissionServiceMock.submitDucrDisassociation(any())(any(), any()))
+          .thenReturn(Future.successful(BAD_REQUEST))
+
+        status(routePost()) must be(INTERNAL_SERVER_ERROR)
+      }
+    }
+
+    def routePost(headers: Map[String, String] = Map.empty, body: JsValue = correctForm): Future[Result] =
+      route(
+        app,
+        FakeRequest(POST, uri)
```

Comment:
Whole this method can go to some helper trait. It's used in few places (or will be used)

Reply:
Created a class to handle routing GET and POST. Thanks for the idea :+1: